### PR TITLE
CITATION.cff: added formatted citation file

### PR DESCRIPTION
**Pull request by neteler**
This PR suggests the CITATION.cff file for improved citation management (by Zenodo.org etc).

The content is based on the GRASS GIS 8.2.1 record: https://doi.org/10.5281/zenodo.7764250

TODO:

- add missing authors relevant for the GRASS 8.2/8.3 releases
- add more ORCID IDs (please verify)

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    email: tomas.zigo@slovanet.sk
+  - given-names: Nicklas
+    family-names: Larsson
+    email: n_larsson@yahoo.com
+  - given-names: Linda
+    family-names: Kladivová
+    email: L.Kladivova@seznam.cz
+  - given-names: Caitlin
+    family-names: Haedrich
+    email: caitlin.haedrich@gmail.com
```
Comment (petrasovaa) [inline CITATION.cff:43]:
```suggestion
  - given-names: Caitlin
    family-names: Haedrich
    email: caitlin.haedrich@gmail.com
    affiliation: North Carolina State University
    orcid: https://orcid.org/0000-0003-4373-5691
```

**Comment by neteler** [comment]
> Any update/sync strategy given that we have contributors file, too? Script in utils would a be start.

I have created a new enhancement ticket #2919 for this suggestion.

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,123 @@
+cff-version: 1.2.0
+message: If you use this software, please cite it using the metadata from this file.
+
+title: GRASS GIS
+version: 8.2.1
```
Comment (neteler) [inline CITATION.cff:5]:
```suggestion
version: 8.3.0
```

File: CITATION.cff [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  - geospatial
+  - spatial analysis
+  - data management
+  - visualization
+  - open source
+  - free software
+  - GNU GPL v2
+
+citation:
+  - text: GRASS Development Team. (2021). Geographic Resources Analysis Support System (GRASS GIS) Software, Version 8.2.1. Open Source Geospatial Foundation. https://grass.osgeo.org
```
Comment (neteler) [inline CITATION.cff:122]:
```suggestion
  - text: GRASS Development Team. (2023). Geographic Resources Analysis Support System (GRASS GIS) Software, Version 8.3.0. Open Source Geospatial Foundation. https://grass.osgeo.org
```

**Comment by neteler** [comment]
I'll merge this now, further editions can be done in future PRs.

The update of this file should also become part of the release CI scripts.

**Comment by neteler** [comment]
(backport will happen in #2920)

**Pull request merged**